File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/Header/IdentityTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4oUnit;
using Db4oUnit.Extensions;
using Db4objects.Db4o.Ext;
using Db4objects.Db4o.Internal;
using Db4objects.Db4o.Tests.Common.Header;

namespace Db4objects.Db4o.Tests.Common.Header
{
	public class IdentityTestCase : AbstractDb4oTestCase
	{
		public static void Main(string[] arguments)
		{
			new IdentityTestCase().RunAll();
		}

		public virtual void TestIdentitySignatureIsNotNull()
		{
			Db4oDatabase identity = Db().Identity();
			Assert.IsNotNull(identity.GetSignature());
		}

		/// <exception cref="System.Exception"></exception>
		public virtual void TestIdentityPreserved()
		{
			Db4oDatabase ident = Db().Identity();
			Reopen();
			Db4oDatabase ident2 = Db().Identity();
			Assert.IsNotNull(ident);
			Assert.AreEqual(ident, ident2);
		}

		/// <exception cref="System.Exception"></exception>
		public virtual void TestGenerateIdentity()
		{
			if (IsMultiSession())
			{
				return;
			}
			byte[] oldSignature = Db().Identity().GetSignature();
			GenerateNewIdentity();
			Reopen();
			ArrayAssert.AreNotEqual(oldSignature, Db().Identity().GetSignature());
		}

		private void GenerateNewIdentity()
		{
			((LocalObjectContainer)Db()).GenerateNewIdentity();
		}
	}
}

File: db4oj.tests/src/com/db4o/db4ounit/common/header/IdentityTestCase.java
/* Copyright (C) 2004 - 2006  Versant Inc.  http://www.db4o.com */

package com.db4o.db4ounit.common.header;

import com.db4o.ext.*;
import com.db4o.internal.*;

import db4ounit.*;
import db4ounit.extensions.*;
import db4ounit.extensions.fixtures.*;

public class IdentityTestCase extends AbstractDb4oTestCase {

	public static void main(String[] arguments) {
		new IdentityTestCase().runAll();
	}
	
	public void testIdentitySignatureIsNotNull() {
		Db4oDatabase identity = db().identity();
		Assert.isNotNull(identity.getSignature());
	}

	public void testIdentityPreserved() throws Exception {

		Db4oDatabase ident = db().identity();

		reopen();

		Db4oDatabase ident2 = db().identity();

		Assert.isNotNull(ident);
		Assert.areEqual(ident, ident2);
	}

	public void testGenerateIdentity() throws Exception {
		if(isMultiSession()){
			return;
		}

		byte[] oldSignature = db().identity().getSignature();

		generateNewIdentity();

		reopen();

		ArrayAssert.areNotEqual(oldSignature, db().identity().getSignature());
	}

	private void generateNewIdentity() {
		((LocalObjectContainer) db()).generateNewIdentity();
	}
}
